BraveWallet/Crypto/Transactions/TransactionDetailsView.swift

```diff
@@ -12,161 +12,34 @@ import Strings
-        return fiat
-      } else {
-        return "$0.00"
-      }
-    @unknown default:
-      return nil
-    }
-  }
-  
-  /// The gas fee including the fiat for the transaction, ex. `("0.003402 ETC", "$15.57")`
-  private var gasFee: (String, fiat: String)? {
-    let isEIP1559Transaction = info.isEIP1559Transaction
-    let limit = info.ethTxGasLimit
-    let formatter = WeiFormatter(decimalFormatStyle: .gasFee(limit: limit.removingHexPrefix, radix: .hex))
-    let hexFee = isEIP1559Transaction ? (info.txDataUnion.ethTxData1559?.maxFeePerGas ?? "") : info.ethTxGasPrice
-    if let value = formatter.decimalString(for: hexFee.removingHexPrefix, radix: .hex, decimals: Int(networkStore.selectedChain.decimals)) {
-      return (value, {
-        guard let doubleValue = Double(value), let assetRatio = assetRatios[networkStore.selectedChain.symbol.lowercased()] else {
-          return "$0.00"
-        }
-        return currencyFormatter.string(from: NSNumber(value: doubleValue * assetRatio)) ?? "$0.00"
-      }())
-    }
-    return nil
-  }
-  
-  /// The formatted transaction fee for the transaction, ex. `"0.003401\n$15.57"`
-  private var transactionFee: String? {
-    guard let (fee, fiat) = gasFee else {
-      return nil
-    }
-    let symbol = networkStore.selectedChain.symbol
-    return String(format: "%@ %@\n%@", fee, symbol, fiat)
-  }
-  
-  /// The market price for the asset
-  private var marketPrice: String {
-    let symbol = networkStore.selectedChain.symbol
-    let marketPrice = currencyFormatter.string(from: NSNumber(value: assetRatios[symbol.lowercased(), default: 0])) ?? "$0.00"
-    return marketPrice
-  }
-
-  private func namedAddress(for address: String) -> String {
-    NamedAddresses.name(for: address, accounts: keyringStore.allAccounts)
-  }
-
-  private var title: String {
-    switch info.txType {
-    case .erc20Approve:
-      return Strings.Wallet.transactionUnknownApprovalTitle
-    default:
-      return info.isSwap ? Strings.Wallet.swap : Strings.Wallet.sent
-    }
-  }
 
   private var header: some View {
     TransactionHeader(
-      fromAccountAddress: info.fromAddress,
-      fromAccountName: namedAddress(for: info.fromAddress),
-      toAccountAddress: info.ethTxToAddress,
-      toAccountName: namedAddress(for: info.ethTxToAddress),
-      originInfo: info.originInfo,
-      transactionType: title,
-      value: value,
-      fiat: fiat
+      fromAccountAddress: transactionDetailsStore.parsedTransaction?.fromAddress ?? "",
+      fromAccountName: transactionDetailsStore.parsedTransaction?.namedFromAddress ?? "",
+      toAccountAddress: transactionDetailsStore.parsedTransaction?.toAddress ?? "",
+      toAccountName: transactionDetailsStore.parsedTransaction?.namedToAddress ?? "",
+      originInfo: transactionDetailsStore.parsedTransaction?.transaction.originInfo,
+      transactionType: transactionDetailsStore.title ?? "",
+      value: transactionDetailsStore.value ?? "",
```

Comment:
Something to think about for later: maybe we should allow passing nil in and then altering the UI when they're empty to not show certain things.  A lot of nil coalescing that would technically end up making the UI look weird if they ever happened